test/e2e/dataplane/service_discovery.go

```diff
@@ -84,7 +92,18 @@ func RunServiceDiscoveryLocalTest(f *framework.Framework) {
 
 	f.DeleteService(framework.ClusterB, nginxServiceClusterB.Name)
 
-	verifyClusterIpWithDig(f, framework.ClusterB, nginxServiceClusterC, netshootPodList, true)
+	cmd := []string{"curl", nginxServiceClusterC.Name}
+	stdout, _, err := f.ExecWithOptions(framework.ExecOptions{
+		Command:       cmd,
+		Namespace:     f.Namespace,
+		PodName:       netshootPodList.Items[0].Name,
+		ContainerName: netshootPodList.Items[0].Spec.Containers[0].Name,
+		CaptureStdout: true,
+		CaptureStderr: true,
+	}, framework.ClusterB)
+
+	Expect(err).NotTo(HaveOccurred())
+	Expect(stdout).To(ContainSubstring("Welcome to nginx!"))
```

Comment:
@vthapar as discussed over slack, please add a log which dumps the IP address returned by LH